### PR TITLE
Hiawatha: Init at 10.5

### PR DESCRIPTION
**Pull request by ndowens**
###### Motivation for this change
Add hiawatha to repo

###### Things done

- [ x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x ] Tested execution of all binary files (usually in `./result/bin/`)
- [ x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -13738,6 +13738,8 @@ with pkgs;
 
   hexedit = callPackage ../applications/editors/hexedit { };
 
+  hiawatha = callPackage ../servers/http/hiawatha { };
```
Comment (cillianderoiste) [inline pkgs/top-level/all-packages.nix]: Could you move this to the "### SERVERS" section instead, please: 
https://github.com/ndowens/nixpkgs/blob/eb232b63ee6b6bb5d2b437694afa0ad1f2617c33/pkgs/top-level/all-packages.nix#L10239
After hbase:
https://github.com/ndowens/nixpkgs/blob/eb232b63ee6b6bb5d2b437694afa0ad1f2617c33/pkgs/top-level/all-packages.nix#L10410
Thanks!

File: pkgs/servers/http/hiawatha/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+               ( if enableTomahawk then "-DENABLE_TOMAHAWK=on" else "-DENABLE_TOMAHAWK=off" )
+               ( if enableXSLT then "-DENABLE_XSLT=on" else "-DENABLE_XSLT=off" )
+               ( if enableToolkit then "-DENABLE_TOOLKIT=on" else "-DENABLE_TOOLKIT=off" )
+               
+             ];
+
+meta = with stdenv.lib; {
+  description = "An advanced and secure webserver";
+  license = licenses.gpl2;
+  homepage = "https://www.hiawatha-webserver.org";
```
Comment (cillianderoiste) [inline pkgs/servers/http/hiawatha/default.nix]: Could you add yourself as a maintainer? You need to include your details in /lib/maintainers.nix too.

Comment (cillianderoiste) [inline pkgs/servers/http/hiawatha/default.nix]: Ah, I see you're already in maintainers.nix :)

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (ndowens) [inline pkgs/top-level/all-packages.nix]: I will once I get home

File: pkgs/servers/http/hiawatha/default.nix (hunk 2 above)
Comment (ndowens) [inline pkgs/servers/http/hiawatha/default.nix]: Yeah I am in the list already 😀 Surprised I left out that line.

**Comment by ndowens** [comment]
Gonna close it and open another; messed up with git

**Pull request closed without merging**